src/main/java/org/truffleruby/core/array/ArraySliceNode.java

```diff
@@ -41,7 +41,9 @@ public DynamicObject readInBounds(DynamicObject array,
         if (emptyArray.profile(from >= to)) {
             return createArray(null, 0);
         } else {
-            final Object store = strategy.newMirror(array).extractRange(from, to).getArray();
+            final ArrayMirror arrayMirror = strategy.newMirror(array);
+            final Object store = arrayMirror.extractRange(from, to).getArray();
+            strategy.setStore(array, arrayMirror.extractRange(0, strategy.getSize(array)).getArray());
```

Comment:
`store` is unused.

Reply:
It's used in the next line.

Reply:
Okay. It was maybe it was wrong to bundle the slice changes in with this lot. When `extractRange` is switched over return a `DelegatedArrayMirror` then the original array must also be switched, otherwise changes to it could be seen in the extracted range.